### PR TITLE
bg/ck_gfx_white_list :  start using ck_utility::is_ck_whitelist to restrict 

### PR DESCRIPTION
**Pull request by bghimireamd**
This PR restricts all CK solvers to MI100, MI200 and MI300.

### CODE REVIEW

File: test/gtest/bn.hpp [hunk 1]
```diff
@@ -177,11 +192,15 @@ struct BNFwdTrainTest
...
     void SetUp() override
     {
-        test_skipped                       = false;
         std::tie(bn_config, tensor_layout) = GetParam();
         bn_fwd_train_test_data.SetUpImpl(bn_config, tensor_layout);
 
         auto&& handle = get_handle();
+        if(!miopen::solver::ck_utility::is_ck_whitelist(handle.GetStream()))
+        {
+            test_skipped = true;
```
Comment (JehandadKhan) [inline test/gtest/bn.hpp:199]:
```suggestion
            GTEST_SKIP() << "Not Applicable on this Architecture";
```

File: test/gtest/bn.hpp [hunk 2]
```diff
@@ -214,7 +233,10 @@ struct BNFwdTrainTest
     void TearDown() override
     {
         if(test_skipped)
```
Comment (JehandadKhan) [inline test/gtest/bn.hpp:233]: Please use the appropriate method in `gtest::testing` namespace to determine if the test was skipped.

File: test/gtest/bn.hpp [hunk 3]
```diff
@@ -129,7 +141,10 @@ struct BNBwdTest : public ::testing::TestWithParam<std::tuple<BNTestCase, miopen
...
+        {
+            GTEST_SKIP() << " Not Applicable : " << bn_config;
```
Comment (JehandadKhan) [inline test/gtest/bn.hpp]: Same as below

**Review by junliume — APPROVED** [review]

File: src/include/miopen/solver/ck_utility_common.hpp [hunk 4]
```diff
@@ -46,6 +46,8 @@ namespace miopen {
 namespace solver {
 namespace ck_utility {
 
+// Disclaimer: Currently CK is only supported in MI100, MI200 and MI300.
+//             Please use is_ck_whitelist instead of this function.
```
Comment (atamazov) [inline src/include/miopen/solver/ck_utility_common.hpp:50]: ⚓

**Pull request merged**